profiler/options.go

```diff
@@ -475,13 +469,23 @@ func WithHTTPClient(client *http.Client) Option {
 
 // WithUDS configures the HTTP client to dial the Datadog Agent via the specified Unix Domain Socket path.
 func WithUDS(socketPath string) Option {
-	return WithHTTPClient(&http.Client{
-		Transport: &http.Transport{
-			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
-				return net.Dial("unix", socketPath)
+	return func(c *config) {
+		// The HTTP client needs a valid URL. The host portion of the
+		// url in particular can't just be the socket path, or else that
+		// will be interpreted as part of the request path and the
+		// request will fail.  Clean up the path here so we get
+		// something resembling the desired path in any profiler logs.
+		// TODO: copied from ddtrace/tracer, but is this correct?
```

Comment:
It looks good enough, as these values tend to interfere in paths.

Reply:
I'm actually not sure I understand this comment. In particular:

1. What was copied from ddtrace/tracer? The path cleaning seems to be unique to the profiler?
2. The path cleaning is only for the log messages right? I.e. `c.agentURL` has no impact on the UDS communication at all?

Maybe the comment can clarify these two things a little?

Reply:
This is what I copied from the tracer:

```
cleanPath := fmt.Sprintf("UDS_%s", strings.NewReplacer(":", "_", "/", "_", `\`, "_").Replace(socketPath))
```

As the comment above indicates, the HTTP client consumes `c.agentURL`. The host portion of the URL is not used in the UDS case since the transport is modified to always connect to the Unix domain socket, but the HTTP client needs to provide the correct path to reach the agent's profile proxy. So we need to at least strip out the `/` characters from the host. I am less sure about the other character replacements, though. I had briefly skimmed https://url.spec.whatwg.org/#valid-host-string to see what it might take to fully sanitize the socket path for use as a host in the URL, but it seemed too complicated...

Reply:
I updated the UDS test case in [b8a865b](https://github.com/DataDog/dd-trace-go/pull/2719/commits/b8a865bd9333319b213f6ae8f53eb08db3dff196) to check that profile uploads go to the right path in the UDS case, to make sure we don't use the filesystem path to the socket as part of the request.